battery_param_range_handler.py

```diff
@@ -0,0 +1,36 @@
+from sys import maxsize
+from Constant import BATTERY_PARAM_CHARACTERISTICS, PARAM_OPERATING_RANGE_CLASSIFIER
+from math import ceil, floor
+
+
+def snap_range_limits_to_sensor_accuracy(battery_param_name, battery_param_operating_ranges):
+    battery_param_sensor_accuracy = BATTERY_PARAM_CHARACTERISTICS[battery_param_name]['sensor_accuracy']
+    for operating_range in battery_param_operating_ranges.values():
+        operating_range[0] = battery_param_sensor_accuracy * ceil(operating_range[0] /
+                                                                  battery_param_sensor_accuracy)
+        operating_range[1] = battery_param_sensor_accuracy * floor(operating_range[1] /
+                                                                   battery_param_sensor_accuracy)
+    return battery_param_operating_ranges
+
+
+def classify_operating_ranges(battery_param_name):
+    param_operating_range = {}
+    param_characteristics = BATTERY_PARAM_CHARACTERISTICS[battery_param_name]
+    tolerance_correction = param_characteristics["tolerance"] * param_characteristics["sensor_limits"]["max"]
+    param_operating_range[PARAM_OPERATING_RANGE_CLASSIFIER[2]] = [-maxsize, param_characteristics["sensor_limits"]["min"]]
```

Comment:
It's hard to correlate the index to the item. Why not use a dictionary?

Reply:
I still have some sort of internal instinct that list is more efficient than a dictionary which might not amount to much I guess for Python. Point noted and I will correct this.